### PR TITLE
JBIDE-19820 Batch feature makes wildfly import way slower

### PR DESCRIPTION
**Pull request by scabanovich**

### CODE REVIEW

**Comment by maxandersen** [comment]
could you add info to the commit message about what this change actually does ?

File: batch/plugins/org.jboss.tools.batch.core/src/org/jboss/tools/batch/core/BatchCorePlugin.java [hunk 1]
```diff
@@ -18,11 +18,11 @@
 import org.jboss.tools.batch.internal.core.impl.BatchProject;
 import org.jboss.tools.batch.internal.core.impl.BatchProjectFactory;
 import org.jboss.tools.batch.internal.core.scanner.BatchArchiveDetector;
+import org.jboss.tools.foundation.core.plugin.BaseCorePlugin;
 import org.jboss.tools.foundation.core.plugin.log.IPluginLog;
-import org.jboss.tools.foundation.ui.plugin.BaseUIPlugin;
 import org.osgi.framework.BundleContext;
 
-public class BatchCorePlugin extends BaseUIPlugin {
```
Comment (maxandersen) [inline batch/plugins/org.jboss.tools.batch.core/src/org/jboss/tools/batch/core/BatchCorePlugin.java]: is this change from core to UI relevant for the performance improvement ?

File: batch/plugins/org.jboss.tools.batch.core/src/org/jboss/tools/batch/internal/core/scanner/BatchArchiveDetector.java [hunk 2]
```diff
@@ -250,4 +252,24 @@ public static IPackageFragmentRoot findPackageFragmentRoot(String jar, IJavaProj
 		return root;
 	}
 
+	boolean computeIsJarBatchArchive(File jarFile) {
+		try {
+			Indexer indexer = new Indexer();
+			Index index = JandexUtil.createJarIndex(jarFile, indexer);
```
Comment (maxandersen) [inline batch/plugins/org.jboss.tools.batch.core/src/org/jboss/tools/batch/internal/core/scanner/BatchArchiveDetector.java]: where does the index files get stored ?

File: batch/plugins/org.jboss.tools.batch.core/src/org/jboss/tools/batch/core/BatchCorePlugin.java (hunk 1 above)
Comment (scabanovich) [inline batch/plugins/org.jboss.tools.batch.core/src/org/jboss/tools/batch/core/BatchCorePlugin.java]: No, but in its triviality it does not deserve a separate issue.

File: batch/plugins/org.jboss.tools.batch.core/src/org/jboss/tools/batch/internal/core/scanner/BatchArchiveDetector.java (hunk 2 above)
Comment (scabanovich) [inline batch/plugins/org.jboss.tools.batch.core/src/org/jboss/tools/batch/internal/core/scanner/BatchArchiveDetector.java]: It is not stored. I found that for big jars it takes _longer_ to save/load than parse once more. For small jars it does not matter.

**Comment by alexeykazakov** [comment]
Rebased and pushed to master.

**Pull request closed without merging**